### PR TITLE
Migrate `foreign-double-unwind`, `issue-36710` and `foreign-exceptions` `run-make` tests to rmake

### PR DESCRIPTION
**Pull request by Oneirical**
Part of #121876 and the associated [Google Summer of Code project](https://blog.rust-lang.org/2024/05/01/gsoc-2024-selected-projects.html).

~~This is expected to fail CI, I am getting a weird `core dumped` error on `foreign-double-unwind` locally. Posting this to start a discussion.~~

EDIT: I got it, `foreign_double_unwind` is *supposed* to fail in the execution stage, but this wasn't obvious, because the original test was just piping the stdout to `CGREP` (which silently ignores errors).

try-job: aarch64-apple
try-job: armhf-gnu
try-job: test-various
try-job: x86_64-msvc
try-job: x86_64-gnu-llvm-18
try-job: i686-msvc
try-job: dist-i586-gnu-i586-i686-musl

### CODE REVIEW

File: tests/run-make/foreign-double-unwind/Makefile
Comment (Oneirical) [inline tests/run-make/foreign-double-unwind/Makefile:9]: Question: is the exit code of this checked at all? The error message could be intentional (2 exceptions thrown in the C++ code - that's what the test is trying to check)

File: tests/run-make/foreign-double-unwind/rmake.rs
Comment (Oneirical) [inline tests/run-make/foreign-double-unwind/rmake.rs:19]: In that case, this should be `run_fail`? But then, executing it afterwards wouldn't be possible...

**Comment by jieyouxu** [comment]
Question: should this be blocked on one of the `build_native_lib` PRs?

**Comment by Oneirical** [comment]
Disregard the last CI fail, I just forgot to commit the fix. It should pass now.

> Question: should this be blocked on one of the `build_native_lib` PRs?

As far as I am aware, the only other PR that modifies `c_build.rs` is #127872, which adds `build_native_dynamic_lib`, an entirely different function.

**Comment by jieyouxu** [comment]
Please try rerunning aarch64-apple again, maybe weird CI issues?

**Comment by Oneirical** [comment]
(The force push is just a rebase)

Neat! Though I do wonder about that `uname()` function failing to match with the string `Darwin`. It also has checks for `FreeBSD`, `SunOS` and `OpenBSD`, and I wonder if the test could fail on these platforms. If they are part of the testing targets at all, that is.

**Comment by jieyouxu** [comment]
> (The force push is just a rebase)
> 
> Neat! Though I do wonder about that `uname()` function failing to match with the string `Darwin`. It also has checks for `FreeBSD`, `SunOS` and `OpenBSD`, and I wonder if the test could fail on these platforms. If they are part of the testing targets at all, that is.

That's possible that the original Makefile was doing a case-insensitive match while I did a case-sensitive match (or vice versa), I'll have to double-check a bit later.

File: tests/run-make/cpp-global-destructors/rmake.rs
Comment (jieyouxu) [inline tests/run-make/cpp-global-destructors/rmake.rs:19]: Discussion:

- I would keep `ignore-none` because, well, no-std.
- I would keep the `wasm` ignores because c++ -> wasm seems... funky.
- I would keep the `sgx` and `nvptx64` because I don't *think* they're tested in full CI (they are tested separately by their target maintainers I think).

File: tests/run-make/foreign-exceptions/rmake.rs
Comment (jieyouxu) [inline tests/run-make/foreign-exceptions/rmake.rs:4]: Suggestion: reword as basic smoke test, this test is not guaranteed to fail if the feature is broken in subtle ways.

**Review by jieyouxu — APPROVED** [review]
Thanks, r=me with a rebase

**Pull request merged**
